node/src/accountant/payment_adjuster/inner.rs

```diff
@@ -1,204 +1,277 @@
         }
     }
 }
 
-impl PaymentAdjusterInner for PaymentAdjusterInnerReal {
-    fn now(&self) -> SystemTime {
-        self.now
-    }
+impl PaymentAdjusterInner {
+    pub fn initialize_guts(
+        &self,
+        tx_count_limit_opt: Option<u16>,
+        cw_service_fee_balance: u128,
+        max_debt_above_threshold_in_qualified_payables_minor: u128,
+    ) {
+        let initialized_guts = GutsOfPaymentAdjusterInner::new(
+            tx_count_limit_opt,
+            cw_service_fee_balance,
+            max_debt_above_threshold_in_qualified_payables_minor,
+        );
 
-    fn largest_exceeding_balance_recently_qualified(&self) -> u128 {
-        self.largest_exceeding_balance_recently_qualified
+        self.initialized_guts_opt
+            .borrow_mut()
+            .replace(initialized_guts);
     }
 
-    fn transaction_fee_count_limit_opt(&self) -> Option<u16> {
-        self.transaction_fee_count_limit_opt
+    pub fn max_debt_above_threshold_in_qualified_payables_minor(&self) -> u128 {
+        self.get_value(
+            "max_debt_above_threshold_in_qualified_payables_minor",
+            |guts_ref| guts_ref.max_debt_above_threshold_in_qualified_payables_minor,
+        )
     }
-    fn original_cw_service_fee_balance_minor(&self) -> u128 {
-        self.original_cw_service_fee_balance_minor
+
+    pub fn transaction_count_limit_opt(&self) -> Option<u16> {
+        self.get_value("transaction_count_limit_opt", |guts_ref| {
+            guts_ref.transaction_count_limit_opt
+        })
     }
-    fn unallocated_cw_service_fee_balance_minor(&self) -> u128 {
-        self.unallocated_cw_service_fee_balance_minor
+    pub fn original_cw_service_fee_balance_minor(&self) -> u128 {
+        self.get_value("original_cw_service_fee_balance_minor", |guts_ref| {
+            guts_ref.original_cw_service_fee_balance_minor
+        })
     }
-    fn subtract_from_unallocated_cw_service_fee_balance_minor(&mut self, subtrahend: u128) {
-        let updated_thought_cw_balance = self
-            .unallocated_cw_service_fee_balance_minor
-            .checked_sub(subtrahend)
-            .expect("should never go beyond zero");
-        self.unallocated_cw_service_fee_balance_minor = updated_thought_cw_balance
+    pub fn remaining_cw_service_fee_balance_minor(&self) -> u128 {
+        self.get_value("remaining_cw_service_fee_balance_minor", |guts_ref| {
+            guts_ref.remaining_cw_service_fee_balance_minor
+        })
     }
-}
-
-#[derive(Default)]
-pub struct PaymentAdjusterInnerNull {}
-
-impl PaymentAdjusterInnerNull {
-    fn panicking_operation(operation: &str) -> ! {
-        panic!(
-            "Broken code: Broken code: Called the null implementation of the {} method in PaymentAdjusterInner",
-            operation
+    pub fn subtract_from_remaining_cw_service_fee_balance_minor(&self, subtrahend: u128) {
+        let updated_thought_cw_balance = self.get_value(
```

Comment:
Let's remove the word `thought` from this name, it turns out that it's not needed.